### PR TITLE
Free up memory in drop box

### PR DESCRIPTION
**Pull request by HerasymovaV**
1. Delete background files when the course is deleted or when the background is changed
2. Delete assignment files when assignment is deleted
3. Delete assignment answers (files) when assignment is deleted

### CODE REVIEW

**Comment by HerasymovaV** [comment]
[Create logic for deletion of files from Dropbox that are attached to the assignments during deletion of course or group](https://trello.com/c/VqLR5R0K/56-create-logic-for-deletion-of-files-from-dropbox-that-are-attached-to-the-assignments-during-deletion-of-course-or-group)

**Comment by HerasymovaV** [comment]
[As a developer, I want to free up memory in dropBox for course background when deleting the course itself](https://trello.com/c/KB9iC0Br/25-as-a-developer-i-want-to-free-up-memory-in-dropbox-for-course-background-when-deleting-the-course-itself)

**Comment by HerasymovaV** [comment]
Don't forget to update database

File: CourseSystem/BLL/Services/CourseBackgroundImageService.cs [hunk 1]
```diff
@@ -0,0 +1,262 @@
...
+        private readonly ILogger<ProfileImageService> _logger;
+
+        public CourseBackgroundImageService(UnitOfWork unitOfWork, IDropboxService dropboxService, ILogger<ProfileImageService> logger)
+            : base(unitOfWork, unitOfWork.CourseBackgroundImageRepository)
+        {
+            _dropboxService = dropboxService;
+            _logger = logger;
+        }
+
+        private bool CheckFileExtension(IFormFile newBackgroundImage)
```
Comment (darhan1168) [inline CourseSystem/BLL/Services/CourseBackgroundImageService.cs:29]: Перенести private метод в конец

File: CourseSystem/Core/ImageStore/DefaultBackgroundImages.cs [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+{
+    private static Dictionary<string, string> _avatarsPack = new Dictionary<string, string>()
+        {
+            {"default-1.png", "https://dl.dropboxusercontent.com/scl/fi/u318svdoeo9mkfbdzf6kl/default-1.png?rlkey=0ustmj3lz8752t6yq03gzaet6&raw=1" },
+            {"default-2.png", "https://dl.dropboxusercontent.com/scl/fi/q69in6su1xo5p5awx4zg3/default-2.png?rlkey=6td6c23r6f7wjvrrbmj1by7mo&raw=1" },
+            {"default-3.png", "https://dl.dropboxusercontent.com/scl/fi/ezuqqrqjfhd7m2d04k8vk/default-3.png?rlkey=iw3se3ffux2vv79qyv045yf2r&raw=1" },
+            {"default-4.png", "https://dl.dropboxusercontent.com/scl/fi/6f74i3t85as6zrpjklkn7/default-4.png?rlkey=qwrrn13d1oz5npi8ua5y488qe&raw=1" }
+        };
+
+    private static Random _random = new Random();
```
Comment (darhan1168) [inline CourseSystem/Core/ImageStore/DefaultBackgroundImages.cs:15]: Отступ

**Review by darhan1168 — APPROVED** [review]

**Review by KatrinaSidorenko — APPROVED** [review]

**Pull request merged**